### PR TITLE
[docker-compose-recipes] LDAP and ClickHouse

### PR DESCRIPTION
**Pull request by nellicus**
this PR implements proper setup of LDAP and ClickHouse.

several LDAP users and groups are mapped to clickhouse Roles with different access levels

### CODE REVIEW

**Comment by DanRoscigno** [comment]
wow, this is great to have!

**Comment by DanRoscigno** [comment]
I will try to test this after the training webinar finishes.

File: docker-compose-recipes/recipes/ch-and-openldap/README.md [hunk 1]
```diff
@@ -2,7 +2,167 @@
...
+All users are using password: `password`
+
+You can validate the LDAP schema using PhpLDAPAdmin UI at http://localhost
+
+Or by running on your host `ldapsearch` command, example:
+
+```
+ldapsearch  -D 'cn=bob,ou=Users,dc=clickhouse,dc=test' -b'dc=clickhouse,dc=test' -H ldap://localhost:389 -w password
+```
+
```
Comment (DanRoscigno) [inline docker-compose-recipes/recipes/ch-and-openldap/README.md:142]:
```suggestion

If you do not have `ldapsearch` on your host you can run it from the openldap container:

```shell
docker compose exec openldap bash -c "ldapsearch  -D 'cn=bob,ou=Users,dc=clickhouse,dc=test' -b'dc=clickhouse,dc=test' -H ldap://localhost:389 -w password"
```

```

File: docker-compose-recipes/recipes/ch-and-openldap/README.md [hunk 2]
```diff
@@ -2,7 +2,167 @@
...
+displayname: LDAP Admin User
+mail: ldapadmin@clickhouse.test
+userPassword:: cGFzc3dvcmQ=
+
+```
+
+These roles are mapped respectively in ClickHouse through the [db](./fs/volumes/clickhouse/docker-entrypoint-initdb.d/1_create_ldap_dbs.sh) and [roles](./fs/volumes/clickhouse/docker-entrypoint-initdb.d/2_create_ldap_roles.sh) configs:
+
+```sql
+CREATE DATABASE sales;
```
Comment (DanRoscigno) [inline docker-compose-recipes/recipes/ch-and-openldap/README.md:90]:
```suggestion
CREATE DATABASE sales_db;
```

File: docker-compose-recipes/recipes/ch-and-openldap/README.md [hunk 3]
```diff
@@ -2,7 +2,167 @@
...
+CREATE DATABASE development;
```
Comment (DanRoscigno) [inline docker-compose-recipes/recipes/ch-and-openldap/README.md:91]:
```suggestion
CREATE DATABASE development_db;
```

**Review by DanRoscigno — CHANGES_REQUESTED** [review]
Love it, tested and added comments.  I have not touched `slapd` since 1997, this brought back bad memories :)

File: docker-compose-recipes/recipes/ch-and-openldap/README.md (hunk 3 above)
Comment (nellicus) [reply docker-compose-recipes/recipes/ch-and-openldap/README.md:91]: pushed changes. FYI this was just the README doc the actual code executing these statements sits in  https://github.com/ClickHouse/examples/pull/30/files#diff-dc21cae1d6a4eaa7975377108a167ea32ed72ab976ebac765eadbb556611ebfd and https://github.com/ClickHouse/examples/pull/30/files#diff-384019e1ea2348ac9f441aa732cac946853041f584102cf929c2d5c14e3bdf3b which I've also changed accordingly

File: docker-compose-recipes/recipes/ch-and-openldap/README.md [hunk 4]
```diff
@@ -2,7 +2,167 @@
...
+                <prefix>clickhouse_</prefix>
+            </role_mapping>
+        </ldap>
+    </user_directories>
+```
+
+All users are using password: `password`
+
+You can validate the LDAP schema using PhpLDAPAdmin UI at http://localhost
+
```
[the remark this answers is not in the document]
Comment (nellicus) [reply docker-compose-recipes/recipes/ch-and-openldap/README.md:136]: pushed changes

**Review by DanRoscigno — APPROVED** [review]

**Pull request merged**